### PR TITLE
V2

### LINKED ISSUE #434: Problem using CLI interface after recent update
### Describe the bug

After a recent update I'm no longer able to use the Nextcloud CLI interface. Some relevant upstream links [here](https://github.com/nextcloud/vm/issues/2039) and [here](https://help.nextcloud.com/t/upgrade-failed-because-of-apcu-now-disabled/120092) but I thought maybe this is YunoHost specific.

```
# sudo -u nextcloud php /var/www/nextcloud/occ app:update --showonly
An unhandled exception has been thrown:
OC\HintException: [0]: Memcache \OC\Memcache\APCu not available for local cache (Is the matching PHP module installed and enabled?)
```

### Context

- Hardware: VPS bought online
- YunoHost version: 4.2.8.1 (stable)
- I have access to my server: Through SSH | through the webadmin
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: yes
  - I installed Debian then added YunoHost, but it hasn't been an issue otherwise (and that was some time ago). But I do remember having trouble with PHP packages some time back. But I thought that was resolved after the Buster migration.
- App version: 21.0.3~ynh1

```
# dpkg -l | grep '^ii' | grep php
ii  php-apcu                              5.1.17+4.0.11-1                       amd64        APC User Cache for PHP
ii  php-apcu-bc                           1.0.4-4                               amd64        APCu Backwards Compatibility Module
ii  php-common                            2:69                                  all          Common files for PHP packages
ii  php-gettext                           1.0.12-0.1                            all          transitional dummy package for php-php-gettext
ii  php-igbinary                          3.0.0-1                               amd64        igbinary PHP serializer
ii  php-imagick                           3.4.3-4.1                             amd64        Provides a wrapper to the ImageMagick library
ii  php-mbstring                          2:7.3+69                              all          MBSTRING module for PHP [default]
ii  php-mysql                             2:7.3+69                              all          MySQL module for PHP [default]
ii  php-pear                              1:1.10.6+submodules+notgz-1.1+deb10u2 all          PEAR Base System
ii  php-php-gettext                       1.0.12-0.1                            all          read gettext MO files directly, without requiring anything other than PHP
ii  php-redis                             4.2.0-1                               amd64        PHP extension for interfacing with Redis
ii  php-smbclient                         0.9.0-3+b1                            amd64        PHP wrapper for libsmbclient
ii  php-tidy                              2:7.3+69                              all          tidy module for PHP [default]
ii  php-xml                               2:7.3+69                              all          DOM, SimpleXML, WDDX, XML, and XSL module for PHP [default]
ii  php7.3-bcmath                         7.3.29-1~deb10u1                      amd64        Bcmath module for PHP
ii  php7.3-bz2                            7.3.29-1~deb10u1                      amd64        bzip2 module for PHP
ii  php7.3-cli                            7.3.29-1~deb10u1                      amd64        command-line interpreter for the PHP scripting language
ii  php7.3-common                         7.3.29-1~deb10u1                      amd64        documentation, examples and common module for PHP
ii  php7.3-curl                           7.3.29-1~deb10u1                      amd64        CURL module for PHP
ii  php7.3-fpm                            7.3.29-1~deb10u1                      amd64        server-side, HTML-embedded scripting language (FPM-CGI binary)
ii  php7.3-gd                             7.3.29-1~deb10u1                      amd64        GD module for PHP
ii  php7.3-gmp                            7.3.29-1~deb10u1                      amd64        GMP module for PHP
ii  php7.3-imap                           7.3.29-1~deb10u1                      amd64        IMAP module for PHP
ii  php7.3-intl                           7.3.29-1~deb10u1                      amd64        Internationalisation module for PHP
ii  php7.3-json                           7.3.29-1~deb10u1                      amd64        JSON module for PHP
ii  php7.3-ldap                           7.3.29-1~deb10u1                      amd64        LDAP module for PHP
ii  php7.3-mbstring                       7.3.29-1~deb10u1                      amd64        MBSTRING module for PHP
ii  php7.3-mysql                          7.3.29-1~deb10u1                      amd64        MySQL module for PHP
ii  php7.3-opcache                        7.3.29-1~deb10u1                      amd64        Zend OpCache module for PHP
ii  php7.3-readline                       7.3.29-1~deb10u1                      amd64        readline module for PHP
ii  php7.3-sqlite3                        7.3.29-1~deb10u1                      amd64        SQLite3 module for PHP
ii  php7.3-tidy                           7.3.29-1~deb10u1                      amd64        tidy module for PHP
ii  php7.3-xml                            7.3.29-1~deb10u1                      amd64        DOM, SimpleXML, WDDX, XML, and XSL module for PHP
ii  php7.3-zip                            7.3.29-1~deb10u1         
```
```
# cat /etc/debian_version 
10.10
```

### Steps to reproduce

```
# sudo -u nextcloud php /var/www/nextcloud/occ app:update --showonly
An unhandled exception has been thrown:
OC\HintException: [0]: Memcache \OC\Memcache\APCu not available for local cache (Is the matching PHP module installed and enabled?)
```

### LINKED ISSUE #540: Create an occ script to easely launch occ command
# Context

On my Yunohost instance, lauching occ command for nextcloud need a long command line :
```bash
sudo -u nextcloud /usr/bin/php8.0 --define apc.enable_cli=1 /var/www/nextcloud/occ ....
```

To avoid this, I made a small script **/usr/bin/occ** :
```bash
#!/bin/bash
# This script allows you to launch occ command according to Nextcloud configuration with Yunohost.

OCC='/var/www/nextcloud/occ'
PHP='/usr/bin/php8.0'
USER='nextcloud'

if [ $( whoami ) != 'root' ]; then
        echo "You need to be root !" >&2
        exit 1
fi

sudo -u $USER $PHP --define apc.enable_cli=1 $OCC $@
```

Do you think it's a good idea to integrate it with the nextcloud app for Yunohost ?

# Environnement
- Debian 5.10.158-2
- Yunohost 11.0.10.2 (stable)
- Nextcloud Hub 3 (25.0.2)

### LINKED ISSUE #591: reboot server, NC is fine for like 20 min then internal server error
### Describe the bug

I want to be as helpful as possible here since obviously I need help, but I might have some issues with knowledge, like I can't find a log for nextcloud in YNH even though the error asks for it. 

Long story short yesterday and in the past all day everyday nextcloud was/is fine. 

I wake up to errors on my phone with syncing calendar so I try to foto the site and get
Internal Server Error

The server encountered an internal error and was unable to complete your request.
Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.
More details can be found in the server log.

So I go down to my server and reboot it. Nextcloud works fine, I go up and shower and come down and check it:
Internal Server Error

The server encountered an internal error and was unable to complete your request.
Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.
More details can be found in the server log.

I reboot again and it's fine. I go make coffee and site down and check it:
Internal Server Error

The server encountered an internal error and was unable to complete your request.
Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.
More details can be found in the server log.

seems every 20 or so min it dies. Everything else YNH is working fine, mattermost, wordpress, vaultwarden, etc. 

### Context

- Hardware: bare metal home pc. 8TB SSD, AMD 5700G, 128GB of RAM
- YunoHost version:11.2.3 (stable)

### PR DESCRIPTION
**Pull request by ericgaspar**
Closes #540 
Closes #434
Closes #591 

- Upgrade package scripts to version 2 -> https://github.com/YunoHost-Apps/nextcloud_ynh/pull/596#issue-1848026616
- #609
- 27.1.3

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by lapineige** [comment]
I don't get what is the failed commit, do you think it's safe enough to test it from current 27.0.2 stable version ?

**Comment by orhtej2** [comment]
I was able to update to this branch with no issues: https://paste.yunohost.org/raw/keyopenumo

Mismatch in `config.php` checksum seems to be a weird side effect of restoring the app prior to this upgrade.

**Comment by ericgaspar** [comment]
There is this line that deactivats the mail app befor upgrading Nextcloud that needs to be fixed... few user failed upgrading Nextcloud because of that.
https://github.com/YunoHost-Apps/nextcloud_ynh/blob/2284f4c34b3cfb985c00621a149c8ade3ec9ac97/scripts/upgrade#L158

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

File: manifest.toml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    [install.user_home]
+    ask.en = "Access the users home folder from Nextcloud?"
+    ask.fr = "Accéder au dossier personnel des utilisateurs depuis Nextcloud ?"
+    type = "boolean"
+    default = false
+
+[resources]
+    [resources.system_user]
+
```
Comment (alexAubin) [inline manifest.toml:56]: cf a question today on the support chat, we may want to enable email in the future, to properly configure email :  

nextcloudemail

(but i don't know much about nextcloud admin, maybe i'm mistaking)

**Comment by ericgaspar** [comment]
!testme

**Pull request merged**